src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,16 +1,100 @@
-import { Component } from 'react';
+import React, { useState } from 'react';
 
-type Props = {};
-type State = {};
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
 
-export class NewMovie extends Component<Props, State> {
-  state: State = {};
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
 
-  render() {
-    return (
-      <form>
-        Put the form here
-      </form>
-    );
-  }
-}
+  const newMovie = {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  };
+
+  const submitHandler = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (title
+      && description
+      && imgUrl
+      && imdbUrl
+      && imdbId) {
```

Comment:
Instead of writing long chains of conditions, try to optimize the code.
```suggestion
    const fields = [title, description, imgUrl, imdbUrl, imdbId];
    if (fields.every(Boolean)) {
```